plugins/by-name/auto-session/default.nix

```diff
@@ -1,121 +1,192 @@
 {
   lib,
-  helpers,
-  config,
-  pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.auto-session;
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
 in
-{
-  options.plugins.auto-session = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "auto-session";
-
-    package = lib.mkPackageOption pkgs "auto-session" {
-      default = [
-        "vimPlugins"
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "auto-session";
+  originalName = "auto-session";
+  package = "auto-session";
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  # TODO: added 204-10-05 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "logLevel"
+    [
+      "sessionLens"
+      "loadOnSetup"
+    ]
+    [
+      "sessionLens"
+      "themeConf"
+    ]
+    [
+      "sessionLens"
+      "previewer"
+    ]
+    [
+      "sessionControl"
+      "controlDir"
+    ]
+    [
+      "sessionControl"
+      "controlFilename"
+    ]
+  ];
+  # NOTE: rename the old settings to use the correct variables
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
         "auto-session"
       ];
-    };
-
-    logLevel = helpers.defaultNullOpts.mkEnum [
-      "debug"
-      "info"
-      "warn"
-      "error"
-    ] "error" "Sets the log level of the plugin.";
-
-    autoSession = {
-      enabled = helpers.defaultNullOpts.mkBool true ''
-        Enables/disables auto creating, saving and restoring.
-      '';
-
-      enableLastSession = helpers.defaultNullOpts.mkBool false ''
-        Whether to enable the "last session" feature.
-      '';
+      settingsPath = basePluginPath ++ [ "settings" ];
+      nestedAutoSessionPluginPath = basePluginPath ++ [ "autoSession" ];
+    in
+    [
+      (lib.mkRenamedOptionModule (
+        basePluginPath
+        ++ [
+          "autoSave"
+          "enabled"
+        ]
+      ) (settingsPath ++ [ "auto_save" ]))
+      (lib.mkRenamedOptionModule (
+        basePluginPath
+        ++ [
+          "autoRestore"
+          "enabled"
+        ]
+      ) (settingsPath ++ [ "auto_restore" ]))
+
+      (lib.mkRenamedOptionModule (
+        basePluginPath
+        ++ [
+          "bypassSessionSaveFileTypes"
+        ]
+      ) (settingsPath ++ [ "bypass_save_filetypes" ]))
+      (lib.mkRenamedOptionModule (nestedAutoSessionPluginPath ++ [ "enableLastSession" ]) (
```

Comment:
Those can't work simply with `optionsRenamedToSettings` ?

Reply:
all of these are changed from a weird nested path to a non nested path for different names. when i was reviewing the documentation it looked like these names matched the https://github.com/rmagatti/auto-session/blob/9d02776ed42874d37869dc683396234e3724b52d/lua/auto-session/config.lua#L120-L151 global variable config names so I'm renaming then to match the https://github.com/rmagatti/auto-session/blob/9d02776ed42874d37869dc683396234e3724b52d/lua/auto-session/config.lua#L63-L107 setup table names

Reply:
Oh right, I read to fast.
Maybe factorizing this with a function could have been a but cleaner, but this is definitely not a blocker.